### ISSUE TITLE
Suspicious binary operation false positive

### ISSUE THREAD

**Issue by kaliberx** [kalessil/phpinspectionsea#1469]
EA Ultimate 3.0.6
Test case:
```php
$x = 1;
$y = 2;
$wrong = ($x === 1) || ($x !== 1 && $y === 2);
$ok = $x === 1 || ($y === 2 && $x !== 1);
```
It reports "[EA] '$x === 1 || $x !== 1' seems to be always true."

**Comment by orklah** [comment]
I think it's correct. You can simplify your expression to 
```php
$wrong = $x === 1 || $y === 2;
```

Because $x will be either === 1 and the first operand will return true, either !== 1 and the second operand depend only of the value of $y.

This is also the case on the "$ok" statement but PHP EA didn't matched this one...

**Comment by kaliberx** [comment]
Thanks for response. In such case you are right, expression can be simplified! Maybe the message is misleading.

What about this example:

```php
$x = 1;
$y = 2;
$a = ($x === 1 && $y === 1) || ($x !== 1 && $y === 2);
$b = ($x === 1 && $y === 1) || ($y === 2);

var_dump($a, $b);
```

Can expression $a be simplified somehow?

**Comment by orklah** [comment]
You're right, in this case it's indeed a false positive.

**Comment by kaliberx** [comment]
I'm using EA Ultimate (personal license), can I expect bug fixed within 5 business days?

**Comment by kalessil** [comment]
@kaliberx: depends on bug rate, but we are running to release frequency (once a month). I'll check if the case is false positive at first place.

**Comment by kalessil** [comment]
Evaluating, I also actualized pricing page - the support time is not realistic currently (we are migrating the licensing solution (JetBrians marketplace), so the whole licensing management and payments are part of JetBrians user account).

**Comment by orklah** [comment]
> Evaluating, I also actualized pricing page - the support time is not realistic currently (we are migrating the licensing solution (JetBrians marketplace), so the whole licensing management and payments will part of JetBrians user account soon).

It would be wise to keep some way to have a subscription outside of the marketplace. In my particular case, my company pays for my JetBrains license but I pay for the patreon subscription. If both are tied on the same invoice, I won't be able to stay a supporter.

**Comment by kalessil** [comment]
Oh, I'll check with folks from JB, how we could address this. Thank you for pointing out!

**Comment by kalessil** [comment]
@kaliberx: the report looks correct to me, so I came up with following solution:

```php
        $x = 1;
        $y = 2;
        $a = ($x === 1 && $y === 1) || ($x !== 1 && $y === 2); // reported
        $b = ($y === 1 && $x === 1) || ($y === 2 && $x !== 1); // ok

        var_dump($a, $b);
``

The change eliminates controversies of left parts of expressions: 
- $y checks are straightforward, so taken as primary condition
- $x related checks became secondary, making the analyzer happy and code less proven to mistakes.
